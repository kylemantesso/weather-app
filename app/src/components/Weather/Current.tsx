import React from "react";
import moment from "moment";

import { Box, Heading, Text } from "grommet";
import { Icon } from "./Icon";
import { Day } from "./Day";
import { FormatType } from "../../context/AppContext";

export const Current: React.FC<{
  description: string;
  currentTemp?: number;
  icon: string;
  windSpeed: string;
  windDirection: string;
  humidity: number;
  formatType: FormatType;
}> = ({
  description,
  currentTemp,
  icon,
  windSpeed,
  windDirection,
  humidity,
  formatType,
}) => (
  <Box direction="row">
    <Box fill direction="row" gap="small">
      <Icon icon={icon} description={description} />
      <Heading size="medium" margin="none">
        {currentTemp && `${currentTemp}Â°${formatType}`}
      </Heading>
    </Box>
    <Box fill>
      <Text>
        Humidity: <Text weight="bold">{Math.round(humidity)}%</Text>
      </Text>
      <Text>
        Wind:{" "}
        <Text weight="bold">
          {windSpeed} {windDirection}
        </Text>
      </Text>
    </Box>
  </Box>
);
