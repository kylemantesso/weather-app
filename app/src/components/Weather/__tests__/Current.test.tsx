import React from "react";
import renderer from "react-test-renderer";
import { Current } from "../Current";

describe("Current", () => {
  it("matches snapshot", () => {
    const component = renderer.create(
      <Current
        currentTemp={27}
        description="Light rain"
        icon="lr"
        formatType="F"
        humidity={80}
        windDirection="S"
        windSpeed="20 kph"
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
