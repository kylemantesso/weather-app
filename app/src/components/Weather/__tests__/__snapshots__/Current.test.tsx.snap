// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current matches snapshot 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 bIdaiR"
>
  <div
    className="StyledBox-sc-13pk1d4-0 iQdAMa"
  >
    <img
      alt="Light rain"
      height="60px"
      src="https://www.metaweather.com/static/img/weather/lr.svg"
    />
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffjnxm"
    />
    <h1
      className="StyledHeading-sc-1rdh4aw-0 bGlFVF"
      size="medium"
    >
      27Â°F
    </h1>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 iNKrsm"
  >
    <span
      className="StyledText-sc-1sadyjn-0 hUokoe"
    >
      Humidity: 
      <span
        className="StyledText-sc-1sadyjn-0 gmtNWw"
      >
        80
        %
      </span>
    </span>
    <span
      className="StyledText-sc-1sadyjn-0 hUokoe"
    >
      Wind:
       
      <span
        className="StyledText-sc-1sadyjn-0 gmtNWw"
      >
        20 kph
         
        S
      </span>
    </span>
  </div>
</div>
`;
